contracts/contracts/MerkleDrop.sol

```diff
@@ -7,38 +7,52 @@ contract MerkleDrop {
 
     bytes32 public root;
     DroppedToken public droppedToken;
+    uint public decayStartTime;
+    uint public decayDurationInSeconds;
 
     mapping (address => bool) withdrawn;
 
     event Withdraw(address recipient, uint value);
 
-    constructor(DroppedToken _droppedToken, bytes32 _root) public {
-        root = _root;
+    constructor(DroppedToken _droppedToken, bytes32 _root, uint _decayStartTime, uint _decayDurationInSeconds) public {
         droppedToken = _droppedToken;
+        root = _root;
+        decayStartTime = _decayStartTime;
+        decayDurationInSeconds = _decayDurationInSeconds;
     }
 
     function withdraw(uint value, bytes32[] memory proof) public {
         withdrawFor(msg.sender, value, proof);
     }
 
     function withdrawFor(address recipient, uint value, bytes32[] memory proof) public {
-        require(checkEntitlement(recipient, value, proof), "The proof could not be verified.");
+        require(verifyEntitled(recipient, value, proof), "The proof could not be verified.");
         require(! withdrawn[recipient], "The recipient has already withdrawn its entitled token.");
-        require(droppedToken.balanceOf(address(this)) > value, "The MerkleDrop does not have tokens to drop yet / anymore.");
+        require(droppedToken.balanceOf(address(this)) >= value, "The MerkleDrop does not have tokens to drop yet / anymore.");
 
         withdrawn[recipient] = true;
-        droppedToken.transfer(recipient, value);
 
+        uint valueToSend = entitlementAtTime(value, now);
```

Comment:
Please add an `assert(valueToSend)<=value` here just to be sure.

Reply:
Done.